lib/sanbase/auth/statistics.ex

```diff
@@ -0,0 +1,103 @@
+defmodule Sanbase.Auth.Statistics do
+  import Ecto.Query
+
+  alias Sanbase.Repo
+  alias Sanbase.Math
+  alias Sanbase.Auth.{User, Settings, UserSettings}
+
+  def tokens_staked() do
+    san_balances =
+      from(u in User)
+      |> positive_san_balance()
+      |> select([u], u.san_balance)
+      |> Repo.all()
+      |> Enum.map(&Decimal.to_float/1)
+
+    %{
+      "average_tokens_staked" => Math.average(san_balances),
+      "median_tokens_staked" => Math.median(san_balances),
+      "tokens_staked" => Enum.sum(san_balances) |> Float.round(2)
+    }
+  end
+
+  @doc ~s"""
+  Return the number of all users that are staking tokens
+  """
+  @spec staking_users() :: non_neg_integer()
+  def staking_users() do
+    from(u in User)
+    |> positive_san_balance()
+    |> count()
+    |> Repo.one()
+  end
+
+  @doc ~s"""
+  Return the number of newly registered users in the given time interval
+  """
+  @spec registered_users(DateTime.t(), DateTime.t()) :: non_neg_integer()
+  def registered_users(%DateTime{} = from, %DateTime{} = to) do
+    registered_users_query(from, to)
+    |> count()
+    |> Repo.one()
+  end
+
+  @doc ~s"""
+  Return the number of newly registered users in the given time interval that have
+  positive SAN balance
+  """
+  @spec registered_staking_users(DateTime.t(), DateTime.t()) :: non_neg_integer()
+  def registered_staking_users(%DateTime{} = from, %DateTime{} = to) do
+    registered_users_query(from, to)
+    |> positive_san_balance()
+    |> count()
+    |> Repo.one()
+  end
+
+  def weekly_updates_subscribed_user_count() do
+    user_settings_with_newsletter_subscription_query(Settings.weekly_subscription_type())
+    |> count()
+    |> Repo.one()
+  end
+
+  def daily_updates_subscribed_user_count() do
+    user_settings_with_newsletter_subscription_query(Settings.daily_subscription_type())
+    |> count()
+    |> Repo.one()
+  end
+
+  # Private functions
+
+  defp user_settings_with_newsletter_subscription_query(subscription_type) do
+    from(us in UserSettings,
+      where:
+        fragment(
+          """
+          settings->>'newsletter_subscription' = ?
+          """,
+          ^subscription_type
+        )
+    )
+  end
+
+  defp registered_users_query(from, to) do
+    from_naive = DateTime.to_naive(from)
+    to_naive = DateTime.to_naive(to)
+
+    from(
+      u in User,
+      where: u.inserted_at >= ^from_naive and u.inserted_at <= ^to_naive
+    )
+  end
+
+  defp positive_san_balance(query) do
```

Comment:
```suggestion
  defp positive_san_balance_query(query) do
```
Lets have a convention for subqueries and to end with `_query` so we know we can plug them. I think you already do it for other subqueries.

Reply:
Now that I think about it if we append `_query` to all such functions it will become weird:
```
 registered_users_query(from, to)
    |> positive_san_balance_query()
    |> count_query()
    |> Repo.one()
```

Another option is to append `query` to the function that initializes the query - the one that does not get another query as an arg

Reply:
My idea is to mark these that return query - so these we know that we can plug them most of the time and they will do what they are supposed to do (filter, aggregate ...)